utils/src/AzExtResourceType.ts

```diff
@@ -1,79 +1,67 @@
-    ApiManagementService = 'ApiManagementService',
-    ApplicationInsights = 'ApplicationInsights',
-    AppServiceKubernetesEnvironment = 'AppServiceKubernetesEnvironment',
-    AppServicePlans = 'AppServicePlans',
-    AvailabilitySets = 'AvailabilitySets',
-    BatchAccounts = 'BatchAccounts',
-    CacheRedis = 'CacheRedis',
-    ContainerRegistry = 'ContainerRegistry',
-    ContainerServiceManagedClusters = 'ContainerServiceManagedClusters',
-    CustomLocations = 'CustomLocations',
-    DeviceIotHubs = 'DeviceIotHubs',
-    DevTestLabs = 'DevTestLabs',
-    Disks = 'Disks',
-    EventGridDomains = 'EventGridDomains',
-    EventGridEventSubscriptions = 'EventGridEventSubscriptions',
-    EventGridTopics = 'EventGridTopics',
-    EventHubNamespaces = 'EventHubNamespaces',
-    FrontDoorAndCdnProfiles = 'FrontDoorAndCdnProfiles',
-    Images = 'Images',
-    KeyVaults = 'KeyVaults',
-    KubernetesConnectedClusters = 'KubernetesConnectedClusters',
-    LoadBalancers = 'LoadBalancers',
-    LogicApp = 'LogicApp',
-    LogicWorkflows = 'LogicWorkflows',
-    ManagedIdentityUserAssignedIdentities = 'ManagedIdentityUserAssignedIdentities',
-    MysqlServers = 'MysqlServers',
-    NetworkApplicationGateways = 'NetworkApplicationGateways',
-    NetworkApplicationSecurityGroups = 'NetworkApplicationSecurityGroups',
-    NetworkInterfaces = 'NetworkInterfaces',
-    NetworkLocalNetworkGateways = 'NetworkLocalNetworkGateways',
-    NetworkPublicIpPrefixes = 'NetworkPublicIpPrefixes',
-    NetworkRouteTables = 'NetworkRouteTables',
-    NetworkSecurityGroups = 'NetworkSecurityGroups',
-    NetworkVirtualNetworkGateways = 'NetworkVirtualNetworkGateways',
-    NetworkWatchers = 'NetworkWatchers',
-    NotificationHubNamespaces = 'NotificationHubNamespaces',
-    OperationalInsightsWorkspaces = 'OperationalInsightsWorkspaces',
-    OperationsManagementSolutions = 'OperationsManagementSolutions',
-    PublicIpAddresses = 'PublicIpAddresses',
-    ServiceBusNamespaces = 'ServiceBusNamespaces',
-    ServiceFabricClusters = 'ServiceFabricClusters',
-    ServiceFabricMeshApplications = 'ServiceFabricMeshApplications',
-    SignalRService = 'SignalRService',
-    SqlDatabases = 'SqlDatabases',
-    SqlServers = 'SqlServers',
-    VirtualMachineScaleSets = 'VirtualMachineScaleSets',
-    VirtualNetworks = 'VirtualNetworks',
-    WebHostingEnvironments = 'WebHostingEnvironments',
-}
-
-/**
- * Gets a normalized type for an Azure resource, accounting for the fact that some
- * Azure resources share values for type and/or kind
- * @param resource The resource to check the {@link AzExtResourceType} for
- * @returns The normalized Azure resource type
- */
-export declare function getAzExtResourceType(resource: { type: string; kind?: string }): AzExtResourceType | undefined;
+
+/**
+ * Normalized type for Azure resources that uniquely identifies resource type for the purposes
+ * of the Azure extensions
+ */
+export enum AzExtResourceType {
```

Comment:
This file shouldn't still be needed since the index contains the declarations

Reply:
I think it's still needed because the [enum exists at runtime](https://www.typescriptlang.org/docs/handbook/enums.html#enums-at-runtime).

Reply:
Ugh...this is why we should be generating our declarations files, not drafting them by hand...

Reply:
> Are there any other enums in index.d.ts that get used elsewhere?

No, this is the first one.

Reply:
> Ugh...this is why we should be generating our declarations files, not drafting them by hand...

I agree. Do you happen to know why we don't generate them? Before I joined, I always thought they were auto generated.

Reply:
Yeah, same here. I think it was just the easiest way to do at the time and we never revisited actually automating it.